lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -582,6 +582,8 @@ static void LoadSwiftFormatters(lldb::TypeCategoryImplSP swift_category_sp) {
                    ConstString("CoreGraphics.CGFloat"), summary_flags);
   AddStringSummary(swift_category_sp, "${var.native}",
                    ConstString("Foundation.CGFloat"), summary_flags);
+  AddStringSummary(swift_category_sp, "${var.native}",
```

Comment:
IIUC, the first parameter of AddStringSummary might be a regex: https://lldb.llvm.org/cpp_reference/FormattersHelpers_8cpp_source.html#l00048

Reply:
`regex` defaults to false.

```
void AddStringSummary(TypeCategoryImpl::SharedPointer category_sp,
                      const char *string, ConstString type_name,
                      TypeSummaryImpl::Flags flags, bool regex = false);
```

Reply:
or are you suggesting I combine the 3 of these into one, with a regex?

Reply:
Maybe we're talking past each other :-)
What I meant was — could we do something like:
```
 AddStringSummary(swift_category_sp, "${var.native}",
                   ConstString("(Foundation|CoreGraphics).CGFloat"), summary_flags, true);
```
?

Reply:
I'm not actually sure this would be practically faster or use less memory, but it seems to be the right thing.

Reply:
I wasn't sure for these reasons:

1. It's unclear whether the existing ones are needed, or can be deleted. I think it's best to keep them because I'm not sure lldb can know for certain which module will be the one exporting this type.
2. Where possible, not breaking the "grep test" is a nice feature (as-is I can grep for "CoreFoundation.CGFloat" but with a regex I can't)
3. For anyone looking at the output of `type summary {list,info}` they'll see a regex instead of a simple type

Reply:
If you prefer regex that's good enough for me

Reply:
I don't think I have a good enough reason to outweigh those. LGTM!

Reply:
For discussion sake, I believe Jim has said that lookup of data formatters prefers exact type names over regex names, which is another (potential/theoretical) argument for exact names.